### PR TITLE
MM-11725: Adding new search_autocomplete channels api

### PR DESCRIPTION
**Pull request by jespino**
#### Summary
Adding new search_autocomplete channels api to the client and to the
actions.

#### Ticket Link
[MM-11725](https://mattermost.atlassian.net/browse/11725)

#### Checklist
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit tests passed
- [x] Ran `make flow` to ensure type checking passed
- [x] Added or updated unit tests (required for all new features)
- [x] All new/modified APIs include changes to the drivers

#### Test Information
This PR was tested on: [Device name(s), OS version(s)]

### CODE REVIEW

**Comment by cpanato** [comment]
@jespino please change to `release-5.3`

**Comment by jespino** [comment]
Delayed 5.4

File: src/actions/channels.js [hunk 1]
```diff
@@ -889,6 +889,37 @@ export function autocompleteChannels(teamId, term) {
...
+        dispatch({type: ChannelTypes.GET_CHANNELS_REQUEST}, getState);
+
+        let channels;
+        try {
+            channels = await Client4.autocompleteChannelsForSearch(teamId, term);
+        } catch (error) {
+            forceLogoutIfNecessary(error, dispatch, getState);
+            dispatch(batchActions([
+                {type: ChannelTypes.GET_CHANNELS_FAILURE, error},
+                logError(error)(dispatch),
```
Comment (enahum) [inline src/actions/channels.js]: this should be `logError(error)` please remove `(dispatch)`

File: test/actions/channels.test.js [hunk 2]
```diff
@@ -1832,6 +1832,22 @@ describe('Actions.Channels', () => {
...
 
+    it('autocompleteChannelsForSearch', async () => {
+        const prefix = TestHelper.basicChannel.name.slice(0, 5);
+
+        nock(Client4.getTeamRoute(TestHelper.basicChannel.team_id)).
+            get('/channels/search_autocomplete').
+            query({name: prefix}).
+            reply(200, [TestHelper.basicChannel]);
+
+        const result = await Actions.autocompleteChannelsForSearch(
```
Comment (enahum) [inline test/actions/channels.test.js]: Please change this
```js
const result = await store.dispatch(Actions.autocompleteChannelsForSearch(
    TestHelper.basicChannel.team_id,
    prefix
));
```

We need to stop doing `action()(dispatch, getState)` as that is an anti-pattern

File: src/actions/channels.js (hunk 1 above)
Comment (jespino) [reply src/actions/channels.js]: Good point. Now fixed.

File: test/actions/channels.test.js (hunk 2 above)
Comment (jespino) [reply test/actions/channels.test.js]: I fixed this for all (dispatch, getState) calls in the channels.test.js

**Comment by jespino** [comment]
@enahum PTAL

**Pull request merged**